src/main/java/com/insightsystems/symphony/dal/dataprobe/DataprobeiBootPDUCommunicator.java

```diff
@@ -423,25 +424,26 @@ private void retrieveMonitoringData() {
 			if (!namesResponse.has("names") || !namesResponse.has("analog")) {
 				throw new Exception("Unable to parse names from response. 'names' field missing in response.");
 			}
-			handleGetDataByResponse(namesResponse.at("/names"), "outletNames", outletNames);
-			handleGetDataByResponse(namesResponse.at("/names"), "groupNames", groupNames);
-			handleGroupGetDataByResponse(namesResponse.at("/analog"), analogProperty);
+			handleGetDataByResponse(namesResponse.at(DataprobeConstant.RESPONSE_NAMES), "outletNames", outletNames);
```

Comment:
you're still pulling namesResponse.at(DataprobeConstant.RESPONSE_NAMES) twice instead of extracting it to JsonNode variable :)